### PR TITLE
Switch Google login system from Flask-OAuthlib to Authlib 

### LINKED ISSUE #1127: Replace Flask-OAuthlib  with Authlib
`cg` is using Flask-Dance and Scout is using Flask-OAuth to handle oauth requests. The latter has been abandoned in favour of Flask-Dance: https://github.com/mitsuhiko/flask-oauth

The problem that arises is that Flask-Oauth requires oauthlib<3.0.0 but Flask-Dance's requirement requests-oauthlib requires oauthlib>=3.0.0.
https://github.com/mitsuhiko/flask-oauth/blob/master/setup.py#L31
https://github.com/requests/requests-oauthlib/blob/master/requirements.txt#L2
Welcome to dependency hell! :)
What are the consequences? Well, for cg it means that all tests are failing unless we take scout-browser away as a dependency.

After investigating this (https://medium.com/knerd/the-nine-circles-of-python-dependency-hell-481d53e3e025), a solution is that scout would upgrade to Flask-Dance at some point. Another is to speed up the uncoupling of cg and scout so you don't have to upgrade.

### PR DESCRIPTION
**Pull request by northwestwitch**
fix #1127 !!

- [x] **How to test it in a cool way locally using your Google email for accessing your demo (localhost:5000) app**:
1. Create credentials for your demo app:

  -  Go to this page:  https://console.developers.google.com/apis/credentials?project=my-project-1507807614573&folder=&organizationId=
  - Click on `credentials` and choose the second option: `OAuth Client ID`
image

  - Fill in this page with the following info: 
 name: whatever
 Authorized JavaScript origins: **https://localhost:5000**
 Authorized redirect URIs: **http://localhost:5000/authorized**
 SAVE
  - Note that a client_id and a client_secret are generated.

  - Edit the scout config file adding the following lines:
 ```
GOOGLE = dict(
    client_id="your_client_id_key",
    client_secret="your_client_secret_key",
    discovery_url="https://accounts.google.com/.well-known/openid-configuration"
)
```

2. Switch to this branch and pip install -r requirements.txt (because they've changed). Try to login with google, using the same user as when you created the credentials above.

3. Test logging in. It shouldn't be possible.

4. Add a user into the demo database that has your google email:
`scout --demo load user -i cust000 -u "Your Name" -m your_email@scilifelab.se`

5. Test logging in now.

- [x] 6. Bonus: test logging in with another gmail account: it shouldn't be possible.

- [x] **How to test it on Stage**:

1. Go to servers config file: https://github.com/Clinical-Genomics/servers/blob/master/config/clinical-db.scilifelab.se/scout-stage.conf.py and modify the GOOGLE login params with the right values, keep in mind that you need just the first 2:
- consumer_key
- consumer_secret
So the new Google login settings will be:
```
GOOGLE = dict(
   client_id="old_consumer_key",
   client_secret="old_consumer_secret",
   discovery_url="https://accounts.google.com/.well-known/openid-configuration"
)
```
- Install this branch on stage
- Test google login
 

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by MM
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
This is done and ready for review. Would you like to do it @moonso ??

File: scout/server/app.py [hunk 1]
```diff
@@ -173,6 +176,25 @@ def cosmic_prefix(cosmicId):
...
+    client_secret = google_conf.get("client_secret")
+
+    extensions.oauth_client.init_app(app)
+
+    extensions.oauth_client.register(
+        name="google",
+        server_metadata_url=discovery_url,
+        client_id=client_id,
+        client_secret=client_secret,
+        client_kwargs={"scope": "openid email profile"},
```
Comment (northwestwitch) [inline scout/server/app.py:194]: No need of all those all params any more, since we are using openid!

File: requirements.txt [hunk 2]
```diff
@@ -21,12 +20,11 @@ pathlib
...
-requests-oauthlib<1.2.0
-oauthlib<3.0.0
-flask_ldap3_login
 
-# unclear
-Flask-Login
+# webapp login
+authlib
+flask_ldap3_login
+flask_login
```
Comment (northwestwitch) [inline requirements.txt:27]: This lib stays, it's the one that is logging the user (regardless if it's obtained from google, ldap or email) in the app

File: requirements.txt [hunk 3]
```diff
@@ -1,12 +1,11 @@
 # Everything needed in production
 
 BeautifulSoup4
-werkzeug<1.0 # This can be removed when switching from Flask-OAuthlib
+werkzeug
```
Comment (moonso) [inline requirements.txt:4]: 👍

File: requirements.txt (hunk 2 above)
Comment (moonso) [reply requirements.txt:27]: Really nice to be able to clean those 😄

File: scout/server/app.py (hunk 1 above)
Comment (moonso) [reply scout/server/app.py:194]: 🥇

**Comment by northwestwitch** [comment]
@moonso I've added some dicumentation. I'll test it on stage now!

**Comment by dnil** [comment]
Nice! Consider a blog post again perhaps?

**Comment by northwestwitch** [comment]
> Nice! Consider a blog post again perhaps?

I'll try, did you see the result of my last attempt? 😆

**Comment by dnil** [comment]
The docs are great, so no worries - a ping will be enough. I just thought they would make a nice post!

**Comment by northwestwitch** [comment]
I've just tested this PR on stage and it works, see here: https://github.com/Clinical-Genomics/servers/pull/371

**Comment by northwestwitch** [comment]
Copying this from a comment from the servers repo. There are other repos with frozen libs because of the old Scout dependencies. We should defrost (thaw?, what's the term in this context?

So I was checking the situation with the libs requirements of the other repos, and I've found out that:
- **cg** has these 3 libs frozen because of scout, which can be updated after we deploy the new scout release, I guess:

image

- **vogue** has also this library that can possibly be unblocked:

image

- **genotype** has the same lib frozen at the moment:

image

- Same thing NIPtTool:

image

**Comment by moonso** [comment]
I'm afraid that those are locked because of other reasons... There was a API change in `werkzeug` when they released `1.0` so I think there are multiple packages that are dependent on that. Can't fully remember.

**Comment by northwestwitch** [comment]
> Can't fully remember.

Me neither

**Comment by northwestwitch** [comment]
Ok, I'm going to merge this, so we can have a new scout release hopefully today!

**Pull request merged**